File: server/src/routes/verification.ts
import { Router } from 'express';
import { query } from '../database/index.js';
import { authenticate, AuthRequest } from '../middleware/auth.js';
import { ValidationError } from '../middleware/errorHandler.js';

const router = Router();

// Get verification timeline for a candidate
router.get('/timeline/:candidateId', authenticate, async (req: AuthRequest, res, next) => {
  try {
    const { candidateId } = req.params;
    const result = await query(
      'SELECT * FROM verification_events WHERE candidate_id = $1 ORDER BY timestamp DESC',
      [candidateId]
    );

    res.json({
      success: true,
      data: result.rows
    });
  } catch (error) {
    next(error);
  }
});

// Create verification event
router.post('/event', authenticate, async (req: AuthRequest, res, next) => {
  try {
    const {
      candidate_id,
      event_type,
      actor,
      details,
      action,
      source_ref
    } = req.body;

    if (!candidate_id || !event_type || !actor || !details) {
      throw new ValidationError('Missing required fields');
    }

    const result = await query(
      `INSERT INTO verification_events (candidate_id, event_type, actor, details, action, source_ref)
      VALUES ($1, $2, $3, $4, $5, $6)
      RETURNING *`,
      [candidate_id, event_type, actor, details, action, source_ref]
    );

    res.status(201).json({
      success: true,
      data: result.rows[0]
    });
  } catch (error) {
    next(error);
  }
});

export { router as verificationRouter };

File: backend/app/routes/verification.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from app.database import get_db
from app.models.user import User
from app.models.verification import VerificationEvent
from app.schemas import ApiResponse
from app.auth import get_current_active_user

router = APIRouter()


@router.get("/candidate/{candidate_id}", response_model=ApiResponse)
async def get_verification_events(
    candidate_id: str,
    current_user: User = Depends(get_current_active_user),
    db: Session = Depends(get_db)
):
    events = db.query(VerificationEvent).filter(
        VerificationEvent.candidate_id == candidate_id
    ).order_by(VerificationEvent.timestamp.desc()).all()
    
    return ApiResponse(
        success=True,
        data=[{
            "id": str(e.id),
            "candidate_id": str(e.candidate_id),
            "timestamp": e.timestamp.isoformat(),
            "event_type": e.event_type,
            "actor": e.actor,
            "details": e.details,
            "action": e.action,
            "source_ref": e.source_ref
        } for e in events]
    )
